plugin/src/main/java/org/opensearch/ml/action/undeploy/TransportUndeployModelAction.java

```diff
@@ -87,19 +93,21 @@ public TransportUndeployModelAction(
 
         this.clusterService = clusterService;
         this.client = client;
+        this.sdkClient = sdkClient;
         this.nodeFilter = nodeFilter;
         this.mlStats = mlStats;
     }
 
     @Override
     protected void doExecute(Task task, MLUndeployModelNodesRequest request, ActionListener<MLUndeployModelNodesResponse> listener) {
         ActionListener<MLUndeployModelNodesResponse> wrappedListener = ActionListener.wrap(undeployModelNodesResponse -> {
-            processUndeployModelResponseAndUpdate(undeployModelNodesResponse, listener);
+            processUndeployModelResponseAndUpdate(request, undeployModelNodesResponse, listener);
         }, listener::onFailure);
         super.doExecute(task, request, wrappedListener);
     }
 
     void processUndeployModelResponseAndUpdate(
+        MLUndeployModelNodesRequest undeployModelNodesRequest,
         MLUndeployModelNodesResponse undeployModelNodesResponse,
         ActionListener<MLUndeployModelNodesResponse> listener
     ) {
```

Comment:
Looks like only additional `tenantId` is needed in this function. Pass the `tenantId` instead of the entire `undeployModelNodesRequest`? As this function is used for processing the response, so passing an entire request is a little bit confusing for me.